src/model_repository_manager.cc

```diff
@@ -594,24 +570,29 @@ ModelRepositoryManager::LoadUnloadModel(
         "explicit load / unload multiple models is not currently supported");
   }
 
-  // Serialize all operations that change model state
-  std::lock_guard<std::mutex> lock(poll_mu_);
-
   const auto& model_name = models.begin()->first;
 
   // Need ModelIdentifier to retrieve model state in lifecycle object,
   // which will not be available after graph update. So make a copy first
   // to check if unload works as intended.
   std::set<ModelIdentifier> to_be_unloaded_ids;
   if (type == ActionType::UNLOAD) {
+    std::lock_guard<std::mutex> lock(mu_);  // guard global_map_
     const auto& git = global_map_.find(model_name);
     if (git != global_map_.end()) {
       to_be_unloaded_ids = git->second;
     }
   }
 
   bool polled = true;
-  RETURN_IF_ERROR(LoadUnloadModels(models, type, unload_dependents, &polled));
+  std::shared_ptr<ConditionVariable> conflict_retry_cv;
```

Comment:
I think you should call this class event listener? If I understand it correctly, `LoadUnloadModels()` set and return this object when the operation can not proceed (not due to error), then the caller will wait for change in state (event trigger, in this case node unlocked).

Reply:
I will call it event notifier, because event listener usually refer to an object that a callback function is provided, which the function will be called when an event happens.

Reply:
Update commit: Rename ConditionVariable to EventNotifier

Reply:
The ConditionVariable / EventNotifier / event listener intermediate class is no longer necessary with some logic simplification.
Update commit: Refactor conflict notify mechanism